File: python_django/core/customer/domain/customer_was_created.py
from core.shared.bus.domain.domain_event import DomainEvent


class CustomerWasCreated(DomainEvent):
    _customer_id: str
    _customer_first_name: str

    def __init__(self, customer_id: str, customer_first_name: str):
        super().__init__()
        self._customer_id = customer_id
        self._customer_first_name = customer_first_name

    def body(self) -> dict:
        return {
            'customerId': self._customer_id,
            'customerFirstName': self._customer_first_name
        }

File: typescript_nestjs/libs/core/src/customer/domain/CustomerWasCreated.ts
import { DomainEvent } from '../../shared/bus/domain/DomainEvent';
import { CustomerPrimitives } from './Customer';

export class CustomerWasCreated extends DomainEvent {
    private _customerId: string;
    private _identification: string;
    private _firstName: string;
    private _lastName: string;
    private _mobilePhone: string;

    constructor({
        customerId,
        identification,
        firstName,
        lastName,
        mobilePhone,
    }: CustomerPrimitives) {
        super();
        this._customerId = customerId;
        this._identification = identification;
        this._firstName = firstName;
        this._lastName = lastName;
        this._mobilePhone = mobilePhone;
    }
}
